.github/workflows/test-suite.yml

```diff
@@ -15,12 +15,12 @@ jobs:
     strategy:
       matrix:
         dependencies:
-          - "locked"
```

Comment:
can you elaborate?

Reply:
I meant that you should upgrade psalm version here: https://github.com/coduo/php-matcher/blob/6.x/tools/composer.json

Reply:
> it's recommended to regenerate the composer.lock file for each PHP version to test.

I'm not sure if I fully agree with that, that's why we have a matrix that is testing this library against lowest/locked/highest dependencies. Who's recommendation this is? Is there any explanation for it somewhere so I can educate myself a bit?  

> Since the file is not present, there are no locked dependencies to test on CI.
That's why this file is committed into the repository.

Reply:
For composer projects, keeping the `composer.lock` is crucial because it provides a snapshot of its dependencies and ensures that the project works the same way in all environments.

For composer libraries, this snapshot can act as a trap. Libraries are supposed to work on multiple platforms, environments, and PHP versions. Also, the library lock file is not used on projects so keeping them in the library provides more cons than pros.

This library is already testing the lowest and the highest versions of the dependencies defined on the `composer.json` file, which implicitly covers any state that could be provided by the lock file.

Reply:
Hi @norberttech I have merged the `6.x` branch into this one to fix merge conflicts.
Let me know if any other change is required. Thank you.

Reply:
Could you pleaase revert removal of composer.lock? Its being also used by dependabot which helps me track updates of dependencies making easier to identify issues when something breaks.

Reply:
@norberttech the PR is ready